packages/cozy-pouch-link/src/PouchManager.js

```diff
@@ -263,6 +281,54 @@ class PouchManager {
   getDatabaseName(doctype) {
     return `${this.options.prefix}_${doctype}`
   }
+
+  async warmupQueries(doctype, queries) {
+    if (!this.warmedUpQueries[doctype]) this.warmedUpQueries[doctype] = []
+    await Promise.all(
+      queries.map(query => {
+        const def = getQueryAlias(query)
+        if (!this.warmedUpQueries[doctype].includes(def)) {
+          this.warmedUpQueries[doctype].push(def)
+          try {
+            return this.executeQuery(query.definition().toDefinition())
+          } catch {
+            delete this.warmedUpQueries[doctype][def]
+          }
+        }
+      })
+    )
+    this.persistwarmedUpQueries()
+    logger.log('PouchManager: warmupQueries for ' + doctype + ' are done')
+  }
+
+  persistwarmedUpQueries() {
+    window.localStorage.setItem(
+      LOCALSTORAGE_WARMUPEDQUERIES_KEY,
+      JSON.stringify(this.warmedUpQueries)
+    )
+  }
+
+  areWarmedUpQueries(doctype, queries) {
+    const persistwarmedUpQueries = this.getPersistedWarmedUpQueries()
+    return queries.every(
+      query =>
+        persistwarmedUpQueries[doctype] &&
+        persistwarmedUpQueries[doctype].includes(getQueryAlias(query))
+    )
+  }
+
+  getPersistedWarmedUpQueries() {
+    const item = window.localStorage.getItem(LOCALSTORAGE_WARMUPEDQUERIES_KEY)
+    if (!item) {
+      return {}
+    }
+    return JSON.parse(item)
+  }
+
+  destroyWarmedUpQueries() {
```

Comment:
I think it's better to keep destroy for async operations ? clear sounds more harmless here and I think is better suited for a cache.